http-client/http-client.cabal

```diff
@@ -97,6 +97,7 @@ test-suite spec-nonet
   build-depends:       base
                      , http-client
                      , hspec
+                     , base-compat
```

Comment:
I'm adding this for the common functions across base versions to read and set environment variables (`lookupEnv`, `setEnv`, `unsetEnv`) so that I could reproduce the bug I found in the test suite. Let me know if there is a way that you think we can get around adding this import.